yewprint/src/numeric_input.rs

```diff
@@ -0,0 +1,200 @@
+            + Clone
+            + Display
+            + FromStr
+            + PartialEq
+            + PartialOrd
+            + Sub<Output = T>
+            + 'static,
+    > Component for NumericInput<T>
+where
+    <T as Sub>::Output: ToString,
+    <T as Add>::Output: ToString,
+{
+    type Message = Msg;
+    type Properties = NumericInputProps<T>;
+
+    fn create(props: Self::Properties, link: ComponentLink<Self>) -> Self {
+        Self {
+            props,
+            link,
+            input: Default::default(),
+        }
+    }
+
+    fn update(&mut self, msg: Self::Message) -> ShouldRender {
+        match msg {
+            Msg::AddEntry => {
+                if let Ok(num) = self.input.trim().parse::<T>() {
+                    self.props.value = Some(num.clone());
+                    self.input = num.to_string();
+                    true
+                } else {
+                    self.props.value = None;
+                    self.input = Default::default();
+                    false
+                }
+            }
+            Msg::UpdateValue(value) => {
+                if let Ok(num) = value.trim().parse::<T>() {
+                    self.props.value = Some(num.clone());
+                    self.input = num.to_string();
+                    true
+                } else {
+                    false
+                }
+            }
+            Msg::Up => {
+                if let Some(num) = self.props.value.clone() {
+                    if num >= self.props.max_value {
+                        self.props.value = Some(self.props.max_value.clone());
+                        self.input = self.props.max_value.to_string();
+                        true
+                    } else {
+                        self.props.value = Some(num.clone() + self.props.increment.clone());
+                        self.input = (num + self.props.increment.clone()).to_string();
+                        true
+                    }
+                } else {
+                    self.props.value = Some(self.props.max_value.clone());
+                    self.input = self.props.max_value.to_string();
+                    true
+                }
+            }
+            Msg::Down => {
+                if let Some(num) = self.props.value.clone() {
+                    if num <= self.props.min_value {
+                        self.props.value = Some(self.props.min_value.clone());
+                        self.input = self.props.min_value.to_string();
+                        true
+                    } else {
+                        self.props.value = Some(num.clone() - self.props.increment.clone());
+                        self.input = (num - self.props.increment.clone()).to_string();
+                        true
+                    }
+                } else {
+                    self.props.value = Some(self.props.min_value.clone());
+                    self.input = self.props.min_value.to_string();
+                    true
```

Comment:
I'm not sure if there should be a valid action to do when your props' value is None...

Putting the minimum seems wrong. Probably using 0 would be better. But then again you don't know if 0 is in the range. Probably the best thing to do here is to do nothing at all.

(Same story in Msg::Up)

```suggestion
                    false
```